typescript/lib/services/service_context.ts

```diff
@@ -0,0 +1,129 @@
+import * as bindingsJsGrpc from '../../kurtosis_core_rpc_api_bindings/api_container_service_grpc_pb'; //TODO - should I be importing all modules or just specific ones for best practices (need to check)
+import * as bindingsJs from '../../kurtosis_core_rpc_api_bindings/api_container_service_pb'; //TODO (comment) - I cannont merge with above line unfortunately, can only import one module at a time MAXIMUM
+import { ServiceID } from './service';
+import { StaticFileID } from './container_creation_config'; 
+import * as grpc from "grpc"; //TODO - this is needed for grpc callback, but this import is tentative 
+import * as constructors from "../constructor_calls";
+var path = require("path"); //TODO - I don't think this works
+
+// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+class GeneratedFileFilepaths { //TODO (comment) - try interface if class doesn't work
+	absoluteFilepathOnTestsuiteContainer: string;
+	absoluteFilepathOnServiceContainer:   string;
+}
+
+class ServiceContext {
+    
+    client: bindingsJsGrpc.ApiContainerServiceClient;
+    serviceId: ServiceID;
+	ipAddress: string;
+	testVolumeMountpointOnTestsuiteContainer: string;
+	testVolumeMountpointOnServiceContainer: string;
+
+    constructor(
+        client: bindingsJsGrpc.ApiContainerServiceClient,
+		serviceId: ServiceID,
+		ipAddress: string,
+		testVolumeMountpointOnTestsuiteContainer: string,
+		testVolumeMountpointOnServiceContainer: string) {
+            this.client = client;
+            this.serviceId = serviceId;
+            this.ipAddress = ipAddress;
+            this.testVolumeMountpointOnTestsuiteContainer = testVolumeMountpointOnTestsuiteContainer;
+            this.testVolumeMountpointOnServiceContainer = testVolumeMountpointOnServiceContainer;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getServiceID(): ServiceID { 
+        return this.serviceId;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getIPAddress(): string {
+        return this.ipAddress;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public execCommand(command: string[]): [number, [], Error] { //TODO - [int32, pointer to []byte (comment), error ] ; is is not []`type` like []string
+        let serviceId = this.serviceId;
+        let args = constructors.newGetExecCommandArgs(serviceId, command);
+        let request: grpc.requestCallback<bindingsJs.ExecCommandResponse>; //TODO - Tried to implement callback through examples on Google while also respecting the actual Callback function inside grpc
```

Comment:
Ooo I'm not sure how this is going to work - my understanding is that this delcaration will initialize it to `null`, and then it will be fed in as `null` to the `client.execCommand`

Given that `this.client.execCommand` looks like it returns a `resp` (rather than a Promise) then maybe you don't need the callback????? Not sure about that one though - you'd have to check the docs for Google's generated Javascript files